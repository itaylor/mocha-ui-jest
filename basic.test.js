const { PNG } = require('pngjs');
const fs = require('fs');

describe('basic test', () => {
  test('expect works', () => {
    expect(true).toBe(true);
    expect([{ that: 'this' }]).toMatchSnapshot();
  });

  test('multiple snapshots in one test work', () => {
    expect({ foo: 'bar' }).toMatchSnapshot();
    expect({ bar: 'foo' }).toMatchSnapshot();
  });

  test('async tests with snapshots work', async () => {
    expect({ before: 'sleep' }).toMatchSnapshot();
    await sleep(10);
    expect({ after: 'sleep' }).toMatchSnapshot();
  });

  test('multiple image assertions work', () => {
    const buf1 = fs.readFileSync(`${__dirname}/fixtures/test-1.png`);
    expect(buf1).toMatchImageSnapshot();
    const buf2 = fs.readFileSync(`${__dirname}/fixtures/test-2.png`);
    expect(buf2).toMatchImageSnapshot();
  });

  test('Image assertions fail by throwing', () => {
    expect(() => {
      expect(makeNewRandomImageBuffer()).toMatchImageSnapshot();
    }).toThrow(/Image snapshot did not match/);
  });
});

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

function makeNewRandomImageBuffer() {
  const dest = PNG.sync.read(fs.readFileSync(`${__dirname}/fixtures/test-1.png`));
  const bomb = PNG.sync.read(fs.readFileSync(`${__dirname}/fixtures/bomb.png`));
  const x = Math.floor(Math.random() * (dest.width - bomb.width));
  const y = Math.floor(Math.random() * (dest.height - bomb.width));
  const img = mergePng(bomb, dest, x, y);
  return PNG.sync.write(img);
}

function mergePng(src, dest, x, y) {
  const { width, height } = src;
  PNG.bitblt(src, dest, 0, 0, width, height, x, y);
  return dest;
}
