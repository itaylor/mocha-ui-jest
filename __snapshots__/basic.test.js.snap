// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot assertions fail by throwing 1`] = `0.9360206344712325`;

exports[`Snapshot assertions work when using expect.any to ignore certain properties 1`] = `
Object {
  "cool": "string",
  "random": Any<Number>,
}
`;

exports[`async tests with snapshots work 1`] = `
Object {
  "before": "sleep",
}
`;

exports[`async tests with snapshots work 2`] = `
Object {
  "after": "sleep",
}
`;

exports[`expect works 1`] = `
Array [
  Object {
    "that": "this",
  },
]
`;

exports[`multiple snapshots in one test work 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`multiple snapshots in one test work 2`] = `
Object {
  "bar": "foo",
}
`;
